src/main/java/net/minecraftforge/fluids/FluidStack.java

```diff
@@ -84,17 +86,46 @@ public static FluidStack loadFluidStackFromNBT(NBTTagCompound nbt)
         {
             return null;
         }
-        if (!nbt.hasKey("FluidName", Constants.NBT.TAG_STRING))
+
+        FluidStack stack;
+        if (nbt.hasKey("Fluid", Constants.NBT.TAG_STRING))
         {
-            return null;
+            ResourceLocation fluid = new ResourceLocation(nbt.getString("Fluid"));
+            if (!ForgeRegistries.FLUIDS.containsKey(fluid))
+            {
+                return null;
+            }
+            else
+            {
+                stack = new FluidStack(ForgeRegistries.FLUIDS.getValue(fluid), nbt.getInteger("Amount"));
+            }
         }
-
-        String fluidName = nbt.getString("FluidName");
-        if (FluidRegistry.getFluid(fluidName) == null)
+        //TODO Drop legacy support for 'FluidNane' in 1.14 or 1.15, as it will be outdated 
+        else if (nbt.hasKey("FluidName", Constants.NBT.TAG_STRING))
+        {
+            String fluidName = nbt.getString("FluidName").toLowerCase();
+            Fluid fluid = null;
+            for (Fluid f : ForgeRegistries.FLUIDS.getValues())
+            {
+                if (f.getRegistryName().getResourcePath().equals(fluidName));
+                {
```

Comment:
I considered that, but it wouldn't make sense to register EVERY Fluid in `FluidDictionary`. Besides, if a modder is making their fluids backwards-compatible, it would make more sense to just lookup a Fluid using `ForgeRegistries.FLUIDS`, since a Fluid with the name forestry:honey in 1.11 MUST have the name forestry:honey in 1.12 and later. So as it has worked in 1.11, the Fluid used for the `FluidStack` would be the first Fluid registered through the `FluidRegistry`, or in 1.12, the first Fluid registered in `ForgeRegistries.FLUIDS`.